osu.Game/Screens/Select/BeatmapInfoWedgeV2.cs

```diff
@@ -0,0 +1,346 @@
+
+        [Resolved]
+        private IBindable<RulesetInfo> ruleset { get; set; } = null!;
+
+        [Resolved]
+        private OsuColour colours { get; set; } = null!;
+
+        protected Container? DisplayedContent { get; private set; }
+
+        protected WedgeInfoText? Info { get; private set; }
+
+        private readonly Container difficultyColourBar;
+        private readonly StarCounter starCounter;
+        private readonly BufferedContainer bufferedContent;
+
+        public BeatmapInfoWedgeV2()
+        {
+            Height = wedge_height;
+            Shear = wedged_container_shear;
+            Masking = true;
+            EdgeEffect = new EdgeEffectParameters
+            {
+                Colour = Colour4.Black.Opacity(.25f),
+                Type = EdgeEffectType.Shadow,
+                Radius = corner_radius,
+                Roundness = corner_radius
+            };
+            CornerRadius = corner_radius;
+
+            // We want to buffer the wedge to avoid weird transparency overlaps between the colour bar and the background.
+            Child = bufferedContent = new BufferedContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Children = new Drawable[]
+                {
+                    // These elements can't be grouped with the rest of the content, due to being present either outside or under the backgrounds area
+                    difficultyColourBar = new Container
+                    {
+                        Colour = Colour4.Transparent,
+                        Depth = float.MaxValue,
+                        Anchor = Anchor.TopRight,
+                        Origin = Anchor.TopRight,
+                        RelativeSizeAxes = Axes.Y,
+
+                        // By limiting the width we avoid this box showing up as an outline around the drawables that are on top of it.
+                        Width = colour_bar_width + corner_radius,
+                        Child = new Box { RelativeSizeAxes = Axes.Both }
+                    },
+                    starCounter = new StarCounter
+                    {
+                        Colour = Colour4.Transparent,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(0.35f),
+                        Shear = -wedged_container_shear,
+                        X = -colour_bar_width / 2,
+                        Direction = FillDirection.Vertical
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            ruleset.BindValueChanged(_ => updateDisplay());
+
+            float starAngle = (float)(Math.Atan(shear_width / wedge_height) * (180 / Math.PI));
+
+            // Applying the rotation directly to the StarCounter distorts the stars, hence it is applied to the child container
```

Comment:
This cannot.

Reply:
and this? if you are unsure how to solve, please say so and I will look into a solution.

Reply:
> and this? if you are unsure how to solve, please say so and I will look into a solution.


I am unsure but mainly I've been a bit busy for a couple of days. A hint would Definitely be appreciated though

Reply:
Put it this way: as a general rule, a drawable should not be interacting with the children of a nested drawable.

The most raw solution would be to fix rotation to work correctly (inside the `StarCounter`). Or make a new derived / new version of the star counter.

Reply:
should be fixed now